ext/ddtrace_profiling_native_extension/collectors_discrete_dynamic_sampler.c

```diff
@@ -36,29 +52,41 @@ static void _discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler, l
     // the next event is automatically sampled by artificially locating it in the interval threshold.
     .events_since_last_sample = BASE_SAMPLING_INTERVAL - 1,
   };
+
+  return NULL;
 }
 
-void discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler) {
+const char* discrete_dynamic_sampler_reset(discrete_dynamic_sampler *sampler) {
   long now = monotonic_wall_time_now_ns(DO_NOT_RAISE_ON_FAILURE);
-  _discrete_dynamic_sampler_reset(sampler, now);
+  return _discrete_dynamic_sampler_reset(sampler, now);
 }
 
-static void _discrete_dynamic_sampler_set_overhead_target_percentage(discrete_dynamic_sampler *sampler, double target_overhead, long now_ns) {
+static const char* _discrete_dynamic_sampler_set_overhead_target_percentage(discrete_dynamic_sampler *sampler, double target_overhead, long now_ns) {
   if (target_overhead <= 0 || target_overhead > 100) {
     rb_raise(rb_eArgError, "Target overhead must be a double between ]0,100] was %f", target_overhead);
   }
   sampler->target_overhead = target_overhead;
-  _discrete_dynamic_sampler_reset(sampler, now_ns);
+  return _discrete_dynamic_sampler_reset(sampler, now_ns);
 }
```

Comment:
Hmmm, this seems a bit inconsistent -- we're reporting the `now_ns` via the new mechanism of returning a string, but if the `target_overhead` doesn't make sense, we raise directly :eyes: